File: datasets/IR-Plag/original-java-rest-kotlin/case-07/plagiarized/L5/01/L5.kt
package IR

import java.util.*

object L5 {
    fun jumlahDiagonal(m: Array<DoubleArray>): Double {
        var sum = 0.0
        var hit = 0

        while (hit != m.size) {
            sum += m[hit][hit]
            hit++
        }
        return sum
    }

    fun print(m: Array<DoubleArray?>?) {
        print("Sum of the elements in the major diagonal is " + jumlahDiagonal(m))
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        val m = Array(4) { DoubleArray(4) }
        println("Enter a 4 by 4 matrix row by row: ")

        for (i in 0..3) {
            for (j in 0..3) {
                m[i][j] = sc.nextDouble()
            }
        }
        L5.print(m)
    }
}

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L5/01/L5.ts
package IR;

import java.util.Scanner;

class L5 {
    static jumlahDiagonal(m: number[][]): number {
        let sum: number = 0;
        let hit: number = 0;

        while (hit !== m.length) {
            sum += m[hit][hit];
            hit++;
        }
        return sum;
    }

    static print(m: number[][]): void {
        console.log(`Sum of the elements in the major diagonal is ${L5.jumlahDiagonal(m)}`);
    }

    static main(args: string[]): void {
        const sc: Scanner = new Scanner(System.in);
        const m: number[][] = new Array(4).fill(0).map(() => new Array(4).fill(0));
        console.log("Enter a 4 by 4 matrix row by row: ");

        for (let i: number = 0; i < 4; i++) {
            for (let j: number = 0; j < 4; j++) {
                m[i][j] = sc.nextDouble();
            }
        }
        L5.print(m);
    }
}

